src/migrations/deploy-stark-proxy-v3.ts

```diff
@@ -0,0 +1,84 @@
+import { Interface } from 'ethers/lib/utils';
+
+import {
+  StarkProxyV1__factory,
+  StarkProxyV3,
+  StarkProxyV3__factory,
+} from '../../types';
+import { getDeployerSigner } from '../deploy-config/get-deployer-address';
+import { log } from '../lib/logging';
+import { getRole, waitForTx } from '../lib/util';
+import { Role } from '../types';
+import { deployUpgradeable } from './helpers/deploy-upgradeable';
+
+export async function deployStarkProxyV3({
+  liquidityStakingAddress,
+  merkleDistributorAddress,
+  starkPerpetualAddress,
+  dydxCollateralTokenAddress,
+  ownerRoleAddress,
+}: {
+  liquidityStakingAddress: string,
+  merkleDistributorAddress: string,
+  starkPerpetualAddress: string,
+  dydxCollateralTokenAddress: string,
+  ownerRoleAddress?: string
+}) {
+  log('Beginning Stark Proxy V3 implementation deployment\n');
+  const deployer = await getDeployerSigner();
+  const deployerAddress = deployer.address;
+  log(`Beginning deployment with deployer ${deployerAddress}\n`);
+
+  log('Step 1. Deploy new StarkProxyV1 implementation contract.');
```

Comment:
nit: note that this step deploys the proxy contract, implementation contract, and proxy admin.